File: 1482-how-many-numbers-are-smaller-than-the-current-number/how-many-numbers-are-smaller-than-the-current-number.java
class Solution {
    public int[] smallerNumbersThanCurrent(int[] nums) {
        int l = nums.length;
        int[] sorted = nums.clone();
        Arrays.sort(sorted);
        int[] output = new int[l];

        Map<Integer,Integer> hash = new HashMap<>();

        for (int i = 0; i < l; i++){
            hash.putIfAbsent(sorted[i],i);
        }

        for (int i = 0; i < l; i++){
            output[i] = hash.get(nums[i]);
        }
        return output;
    }
}

File: 1482-how-many-numbers-are-smaller-than-the-current-number/how-many-numbers-are-smaller-than-the-current-number.cs
public class Solution {
    public int[] SmallerNumbersThanCurrent(int[] nums) {
        int l =nums.Length;
        int[] output = new int[l];
        Dictionary<int,int> hash = new Dictionary<int,int>();
        int[] sorted = (int[])nums.Clone();
        Array.Sort(sorted);
        
        for (int i = 0; i < l; i++ ){
            if (!hash.ContainsKey(sorted[i]))
            hash[sorted[i]] = i;
        }

        for (int i = 0; i< l; i++){
            output[i] = hash[nums[i]];
        }
        return output;
    }
}